### ISSUE TITLE
SC1010 False positive whit mlr then

### ISSUE THREAD

**Issue by afranke** [koalaman/shellcheck#1801]
Version 0.7.0

- [x] The rule's wiki page does not already cover this
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
mlr clean-whitespace then sort -f group foo.csv
```

#### Here's what shellcheck currently says:
`^-- SC1010: Use semicolon or linefeed before 'then' (or quote to make it literal).`

#### Here's what I wanted or expected to see:

Nothing as `then` is a [`mlr`](https://johnkerl.org/miller/doc/) keyword.

**Comment by Nightfirecat** [comment]
This seems like a clear case for [ignoring the warning](https://github.com/koalaman/shellcheck/wiki/Ignore) since I don't think an exception will be added for a non-standard cli utility.

**Comment by afranke** [comment]
Well I guess it is your call, but it sure would make life easier for all the users of such tools. One argument in favor of handling it is that with the proposed ignore methods if I still want to get legitimate reports I have to annotate *every* use of the utility, and some of my scripts are relying on mlr pretty heavily. If I use a script-wise ignore rule or if I use the `.shellcheckrc` method, I will miss out on all the `done` reports and that shell keyword is used quite often.

**Comment by gromgit** [comment]
> This seems like a clear case for [ignoring the warning](https://github.com/koalaman/shellcheck/wiki/Ignore) since I don't think an exception will be added for a non-standard cli utility.

I think it's actually a bug. Why is shellcheck treating `then` as a reserved word without a preceding `if`?

**Comment by TinCanTech** [comment]
> Well I guess it is your call, but it sure would make life easier for all the users of such tools. One argument in favor of handling it is that with the proposed ignore methods if I still want to get legitimate reports I have to annotate _every_ use of the utility, and some of my scripts are relying on mlr pretty heavily. If I use a script-wise ignore rule or if I use the `.shellcheckrc` method, I will miss out on all the `done` reports and that shell keyword is used quite often.

#1417

**Comment by afranke** [comment]
> #1417

Nope. That ticket is about ignoring/not ignoring a whole rule. I was pointing out that ignoring a whole rule is the problem.
